modules/data-entry/src/main/java/io/uhndata/cards/internal/MaxFormsOfTypePerSubjectValidator.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.uhndata.cards.internal;
+
+import javax.jcr.Node;
+import javax.jcr.PropertyIterator;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+
+import org.apache.jackrabbit.oak.api.CommitFailedException;
+import org.apache.jackrabbit.oak.api.Type;
+import org.apache.jackrabbit.oak.spi.commit.DefaultValidator;
+import org.apache.jackrabbit.oak.spi.commit.Validator;
+import org.apache.jackrabbit.oak.spi.state.NodeBuilder;
+import org.apache.jackrabbit.oak.spi.state.NodeState;
+
+public class MaxFormsOfTypePerSubjectValidator extends DefaultValidator
+{
+    private final Session session;
+
+    // This holds the builder for the current node. The methods called for editing specific properties don't receive the
+    // actual parent node of those properties, so we must manually keep track of the current node.
+    private final NodeBuilder currentNodeBuilder;
+
+    public MaxFormsOfTypePerSubjectValidator(Session session, NodeBuilder currentNodeBuilder)
+    {
+        this.session = session;
+        this.currentNodeBuilder = currentNodeBuilder;
+    }
+
+    @Override
+    public Validator childNodeAdded(String name, NodeState after) throws CommitFailedException
+    {
+
+        if (isForm(this.currentNodeBuilder)) {
+            try {
+                Node questionnaire = this.session.getNodeByIdentifier(this.currentNodeBuilder
+                        .getProperty("questionnaire").getValue(Type.REFERENCE));
+                Node subject = this.session.getNodeByIdentifier(this.currentNodeBuilder.getProperty("subject")
+                        .getValue(Type.REFERENCE));
+                PropertyIterator maxFormNumberPerSubject = questionnaire.getProperties("maxPerSubject");
+                if (maxFormNumberPerSubject != null && Integer.parseInt(maxFormNumberPerSubject.toString()) > 0) {
+                    int formNumber = countFormsPerSubject(subject);
```

Comment:
```suggestion
                    int formNumber = countFormsPerSubject(subject) + 1;
```

Add `1` because we want the number of _Forms_ for this _Subject_ that already exist, plus this current new one, right?